tests/opt-testcases/tls13-misc.sh

```diff
@@ -277,6 +277,31 @@ run_test    "TLS 1.3 m->G: EarlyData: basic check, good" \
             -s "END OF EARLY DATA (5) was received." \
             -s "early data accepted"
 
+requires_gnutls_tls1_3
+requires_config_enabled MBEDTLS_DEBUG_C
+requires_config_enabled MBEDTLS_SSL_CLI_C
+requires_all_configs_enabled MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE \
+                             MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_EPHEMERAL_ENABLED \
+                             MBEDTLS_SSL_EARLY_DATA
+requires_any_configs_enabled MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_EPHEMERAL_ENABLED \
+                             MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_ENABLED
+run_test    "TLS 1.3 m->G: EarlyData: write early data, good" \
+            "$G_NEXT_SRV -d 10 --priority=NORMAL:-VERS-ALL:+VERS-TLS1.3:+CIPHER-ALL:+ECDHE-PSK:+PSK --earlydata --disable-client-cert" \
+            "$P_CLI debug_level=4 early_data=1 reco_mode=1 reconnect=1 reco_delay=900" \
+            0 \
+            -c "Reconnecting with saved session" \
+            -c "NewSessionTicket: early_data(42) extension received." \
+            -c "ClientHello: early_data(42) extension exists." \
+            -c "EncryptedExtensions: early_data(42) extension received." \
+            -c "EncryptedExtensions: early_data(42) extension exists." \
+            -c "<= write early_data" \
+            -c "<= write EndOfEarlyData" \
+            -s "Parsing extension 'Early Data/42' (0 bytes)" \
+            -s "Sending extension Early Data/42 (0 bytes)" \
+            -s "END OF EARLY DATA (5) was received." \
+            -s "early data accepted" \
+            -s "decrypted early data with length"
+
```

Comment:
Notes:
- We appear to be only testing early data with GnuTLS, though it seems to be we should be able to test with ourselves too now. (Not expecting any issues, but as matter of principle, would be pretty embarrassing if we couldn't interop with ourselves.)
- In addition to the tests with `early_data=1` I'd like to see two reference tests (one with `early_data=0`, one without `early_data` option - so, library default), with the same list of assertions except now all negative (except "Reconnecting with saved session"), just to validate that we're really not sending it unless explicitly enabled at runtime (even when enabled at compile time. (Again, not expecting this to uncover issues, more as matter of principle.)

Edit: both tracked in #6362, so not a problem for this PR.

Reply:
Added a pointer to this comment in #6362 to be sure we cover it all.